app/models/manageiq/providers/redhat/infra_manager/supported_apis_mixin.rb

```diff
@@ -0,0 +1,59 @@
+module ManageIQ::Providers::Redhat::InfraManager::SupportedApisMixin
+  DEFAULT_PRODUCT_VERSION_TO_API_VERSION_REGEX_HASH = {
+    '^3\..*$' => %w(3),
+    '^4\..*$' => %w(3 4)
+  }.freeze
+
+  def supported_api_versions
+    return supported_api_versions_from_sdk(probe_args) if api_version.blank?
+    supported_api_versions_from_db
+  end
+
+  def highest_supported_api_version
+    supported_api_versions.sort.last || '3'
+  end
+
+  def highest_allowed_api_version
+    return '3' unless use_ovirt_sdk?
+    highest_supported_api_version
+  end
+
+  # This method is a result of api_version in the db not actually being the api_version
+  # it is actually the product version based on which we infer the api version.
+  # In the future I hope there will be a separate field created.
+  def supported_api_versions_from_db
+    product_v_to_api_v_hash = DEFAULT_PRODUCT_VERSION_TO_API_VERSION_REGEX_HASH ||
+                              Setting::ems::ems_redhat.default_product_version_to_api_version_regex_hash
+    key = product_v_to_api_v_hash.keys.detect { |k| api_version =~ Regexp.new(k, Regexp::EXTENDED) }
```

Comment:
Does this guarantee that we will always select the first regex that matches, in the same order that they appear in the `settings.yaml` file? It is important that we make that predictable.

Reply:
It does, although I do not think its a good thing to relay on while writing the regexps, I think they should be accurate enough to avoid confusion.

Reply:
I agree, both things: they should be accurate and they should be processed in a predictable order, same than in the `settings.yaml` file.

Reply:
done :)